concordium-consensus/src/Concordium/GlobalState/Account.hs

```diff
@@ -131,24 +131,28 @@ addIncomingEncryptedAmount newAmount old =
         Nothing ->
             -- we have to aggregate if we have >= maxNumIncoming amounts on the sequence
             if Seq.length (_incomingEncryptedAmounts old) >= maxNumIncoming
-                then -- irrefutable because of check above
-
-                    let ~(x Seq.:<| y Seq.:<| rest) = _incomingEncryptedAmounts old
-                    in  old
+                then case _incomingEncryptedAmounts old of
+                    (x Seq.:<| y Seq.:<| rest) ->
+                        old
                             { _incomingEncryptedAmounts = rest Seq.|> newAmount,
                               _startIndex = _startIndex old + 1,
                               _aggregatedAmount = Just (x <> y, 2)
                             }
+                    -- this does not happen due to the check above
+                    _ -> error "_incomingEncryptedAmounts should consist of two or more elements"
                 else old & incomingEncryptedAmounts %~ (Seq.|> newAmount)
         Just (e, n) ->
             -- we have to aggregate always
-            -- irrefutable because of check above
```

Comment:
Same as my other comment above:

Is this really irrefutable? Looking at the commit history, it seems that some of it was moved outside the check. I blindly added the `error` to the case which preserves the semantics and removes the warning, but maybe it is worth investigating.